### PR TITLE
App preparing

### PR DESCRIPTION
**Pull request by r34son**
Using [feature sliced design](https://feature-sliced.design/) architectural methodology

### CODE REVIEW

File: .husky/pre-commit [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env sh
+. "$(dirname -- "$0")/_/husky.sh"
+
+yarn lint-staged
```
Comment (Arillaxe) [inline .husky/pre-commit:4]: No new line at the end of the file

File: package.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    "@emotion/react": "^11.10.5",
+    "@emotion/server": "^11.10.0",
+    "@emotion/styled": "^11.10.5",
+    "@mui/icons-material": "^5.11.0",
+    "@mui/material": "^5.11.7",
+    "@next/font": "^13.1.6",
+    "@types/node": "18.11.18",
+    "@types/react": "18.0.27",
+    "@types/react-dom": "18.0.10",
+    "eslint": "8.33.0",
```
Comment (Arillaxe) [inline package.json:22]: `eslint` and other types packages should be in devDependencies

File: src/widgets/header/ui/Profile.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  user: {
+    name?: string | null;
+    image?: string | null;
+  };
+}
+
+export const Profile = ({ user }: ProfileProps) => {
+  const [anchorElUser, setAnchorElUser] = useState<HTMLElement | null>(null);
+
+  const handleOpenUserMenu = (event: React.MouseEvent<HTMLElement>) => {
```
Comment (Arillaxe) [inline src/widgets/header/ui/Profile.tsx:15]: image

**Review by Arillaxe — COMMENTED** [review]
Slabo

File: package.json (hunk 2 above)
Comment (r34son) [reply package.json:22]: Sure 👌

File: src/widgets/header/ui/Profile.tsx (hunk 3 above)
Comment (r34son) [reply src/widgets/header/ui/Profile.tsx:15]: Do you think wrapping in useCallback and memoizing component that accepts this function as prop will be more efficient?

**Pull request merged**